### ISSUE TITLE
I2C address of your Qwiic Raspberry Pi Hat?

### ISSUE THREAD

**Issue by ChristianFrisson** [WHC2021SIC/WHC2021SIC-Guide#2]
Hi @WHC2021SIC/sic-students,

We would greatly appreciate your help so that we determine whether one team received faulty hardware, in particular one Qwiic Raspberry Pi Hat.

Here is a very fast test procedure, adapted from the [Qwiic HAT for Raspberry Pi Hookup Guide](https://learn.sparkfun.com/tutorials/qwiic-hat-for-raspberry-pi-hookup-guide/all), assuming you have setup up your kit following our guide including section [SparkFun Qwiic hat, boards and sensors](https://github.com/WHC2021SIC/WHC2021SIC-Guide#sparkfun-qwiic-hat-boards-and-sensors-1), and that would take you 1 minute to complete and give us insight:
- hardware: connect only the Qwiic Raspberry Pi Hat to the Rapberry Pi (disconnect all other Qwiic cables from the hat, and disconnect the Octo soundcard)
- software: in a terminal, copy/paste the following, line-by-line:
```
sudo apt-get install -y i2c-tools
i2cdetect -y 1
```
You will get an output from your Raspberry Pi similar to the output below.
```
pi@raspberrypi:~/$ i2cdetect -y 1
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: 60 -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --
```

Could you share with us the I2C address of your Qwiic Raspberry Pi Hat that appears in the output (in the example: `60`)?

**Comment by ADataDate** [comment]
The Qwiic Pi hat won't have an I2C address associated with it, but if you connect the I2C mux to the Qwiic pi hat you should see the default address of 0x70 assuming you didn't change any jumper settings. 
![image](https://user-images.githubusercontent.com/16926827/120958971-001c6180-c716-11eb-9071-96c4bb5efcfa.png)

I am curious about the "UU". It appears in the same location as the 0x48 and It only appears when the octo sound card is also connected. I looked up that code and according to this forum (https://linux.die.net/man/8/i2cdetect) probing was skipped because it's in use. I couldn't find any octo driver info to confirm this.

**Comment by ADataDate** [comment]
Not directly related to this issue, and maybe I'll open a separate issue, but building syntacts on Linux was a struggle, and I'm not entirely sure it built correctly. I tested a couple of the examples from the python and C example libraries, but they all behaved similarly. The number of changes necessary to the CMAKE file and syntacts.py to find the Linux DLL (shared objects) was difficult to figure out (not sure I did) - on top of finding all the right development files.  
![image](https://user-images.githubusercontent.com/16926827/120959569-49b97c00-c717-11eb-8b8e-5fe180994ca1.png)

I had to add the ubuntu platform check to this and direct it to the downloaded files.

**Comment by ChristianFrisson** [comment]
> The Qwiic Pi hat won't have an I2C address associated with it, but if you connect the I2C mux to the Qwiic pi hat you should see the default address of 0x70 assuming you didn't change any jumper settings.
> ![image](https://user-images.githubusercontent.com/16926827/120958971-001c6180-c716-11eb-9071-96c4bb5efcfa.png)

Thanks @ADataDate for confirming that the I2C reading is not relevant if only the Qwiic Hat is connected!

> I am curious about the "UU". It appears in the same location as the 0x48 and It only appears when the octo sound card is also connected. I looked up that code and according to this forum (https://linux.die.net/man/8/i2cdetect) probing was skipped because it's in use. I couldn't find any octo driver info to confirm this.

Here is the list of GPIO pins used by the Octo hat: http://forum.audioinjector.net/viewtopic.php@f=9&t=1317.html 0x48 is not listed there. Is this issue blocking you project?

**Comment by ChristianFrisson** [comment]
> Not directly related to this issue, and maybe I'll open a separate issue, but building syntacts on Linux was a struggle, and I'm not entirely sure it built correctly. I tested a couple of the examples from the python and C example libraries, but they all behaved similarly. The number of changes necessary to the CMAKE file and syntacts.py to find the Linux DLL (shared objects) was difficult to figure out (not sure I did) - on top of finding all the right development files.
> ![image](https://user-images.githubusercontent.com/16926827/120959569-49b97c00-c717-11eb-8b8e-5fe180994ca1.png)
> 
> I had to add the ubuntu platform check to this and direct it to the downloaded files.

Yes that would be great if we continue this discussion in another issue here: https://github.com/WHC2021SIC/Syntacts/issues and if you could share your changes in a new branch https://github.com/WHC2021SIC/Syntacts/branches ! 

I have just updated the settings of https://github.com/WHC2021SIC/Syntacts: all sic-students should have write access now.

**Comment by ADataDate** [comment]
> > The Qwiic Pi hat won't have an I2C address associated with it, but if you connect the I2C mux to the Qwiic pi hat you should see the default address of 0x70 assuming you didn't change any jumper settings.
> > ![image](https://user-images.githubusercontent.com/16926827/120958971-001c6180-c716-11eb-9071-96c4bb5efcfa.png)
> 
> Thanks @ADataDate for confirming that the I2C reading is not relevant if only the Qwiic Hat is connected!
> 
> > I am curious about the "UU". It appears in the same location as the 0x48 and It only appears when the octo sound card is also connected. I looked up that code and according to this forum (https://linux.die.net/man/8/i2cdetect) probing was skipped because it's in use. I couldn't find any octo driver info to confirm this.
> 
> Here is the list of GPIO pins used by the Octo hat: http://forum.audioinjector.net/viewtopic.php@f=9&t=1317.html 0x48 is not listed there. Is this issue blocking you project?

This is not blocking anything but I noticed the location of the "UU" is in the same location as another team that is reading 0x48.

I'll post an update in the new issue and push changes to the new branch here shortly - thank you for creating those!
